tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -24,9 +24,262 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
+import java.lang.reflect.Array;
+import java.nio.*;
+import java.util.Arrays;
+
 /** Unit tests for {@link org.tensorflow.Tensor}. */
 @RunWith(JUnit4.class)
 public class TensorTest {
+  private static final double EPSILON = 1e-7;
+  private static final float EPSILON_F = 1e-7f;
+
+  // reusable sample data, varying by type and dimension
```

Comment:
Not all of these are re-used between tests (for example, the shape ones are needed only for the `create` tests).

I'd suggest doing away with class variables and instead defining these within the test methods. That may come at the cost of some duplication, but the resulting tests are more readable if they are self contained.